### PR TITLE
[XLA:GPU][ROCm] Enabling amdgpu backend in XLA unit tests

### PR DESCRIPTION
**Pull request by jerryyin**
This PR enables amdgpu XLA backend for unit tests, in preparation for setting up ROCm XLA community support builds.

### CODE REVIEW

File: tensorflow/compiler/xla/tests/build_defs.bzl [hunk 1]
```diff
@@ -24,6 +26,8 @@ def filter_backends(backends):
     """
     if cuda_is_configured():
```
Comment (cheshire) [inline tensorflow/compiler/xla/tests/build_defs.bzl]: `cuda_is_configured() or rocm_is_configured()`?

File: tensorflow/compiler/xla/tests/build_defs.bzl [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
     "//tensorflow/core/platform:default/build_config_root.bzl",
     "tf_cuda_tests_tags",
+    "tf_gpu_tests_tags",
 )
 
 all_backends = ["cpu", "gpu"] + plugins.keys()
 
 def filter_backends(backends):
-    """Removes "gpu" from a backend list if CUDA is not enabled.
+    """Removes "gpu" from a backend list if Cuda or Rocm is not enabled.
```
Comment (cheshire) [inline tensorflow/compiler/xla/tests/build_defs.bzl]: CUDA

**Review by cheshire — CHANGES_REQUESTED** [review]
LGTM apart from minor nits

**Review by cheshire — APPROVED** [review]

**Comment by gbaned** [comment]
@jerryyin Could you please check failed build errors? Thanks!

**Comment by jerryyin** [comment]
@gbaned Judging from the build failures: `external/local_config_mlir/include/mlir/Dialect/QuantOps/UniformSupport.h:148:14: error: 'clamp' is not a member of 'std'`

I don't think the failures have anything to do with this PR. `std::clamp` is a C++ 17 addition and shouldn't be used in that header file.

**Pull request merged**